http/oidc/src/test/java/org/wildfly/security/http/oidc/OidcTest.java

```diff
@@ -199,6 +231,52 @@ private InputStream getOidcConfigurationInputStream(String clientSecret) {
         return getOidcConfigurationInputStream(clientSecret, KEYCLOAK_CONTAINER.getAuthServerUrl());
     }
 
+    private void checkScopeValuesInHTTPServerResponse(String expectedScope) throws HttpAuthenticationException, URISyntaxException {
+        Map<String, Object> props = new HashMap<>();
+        HttpServerAuthenticationMechanism mechanism = oidcFactory.createAuthenticationMechanism(OIDC_NAME, props, getCallbackHandler());
+        URI requestUri = new URI(getClientUrl());
+        TestingHttpServerRequest request = new TestingHttpServerRequest(null, requestUri);
+        mechanism.evaluateRequest(request);
+        TestingHttpServerResponse response = request.getResponse();
+        assert(response.getFirstResponseHeaderValue("Location").contains("scope=" + expectedScope));
+    }
+
+    private void checkInvalidScopeError(InputStream oidcConfig, String username, String password, boolean loginToKeycloak,
```

Comment:
This method also contains a lot of duplicate code from `performAuthentication`. Just wondering if we could maybe tweak `performAuthentication` in a way similar to what I described above to cover this case too?

Reply:
Will do!

Reply:
All comments up to this one have been addressed!